x-pack/qa/saml-idp-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticationIT.java

```diff
@@ -131,12 +148,24 @@ public class SamlAuthenticationIT extends ESRestTestCase {
         .setting("xpack.security.authc.realms.native.native.order", "4")
         .setting("xpack.ml.enabled", "false")
         .setting("logger.org.elasticsearch.xpack.security", "TRACE")
-        .configFile("sp-signing.key", Resource.fromClasspath("sp-signing.key"))
-        .configFile("idp-metadata.xml", Resource.fromClasspath("idp-metadata.xml"))
-        .configFile("sp-signing.crt", Resource.fromClasspath("sp-signing.crt"))
+        .configFile("sp-signing.key", Resource.fromClasspath("/idp/shibboleth-idp/credentials/sp-signing.key"))
+        .configFile("idp-metadata.xml", () -> Resource.fromString(calculateIdpMetaData()))
```

Comment:
I believe we already have a mechanism for lazily evaluated resources that avoids all the extra complexity of adding a `Supplier<Resource>` method to the builder API. This could be simplified to `Resource.fromString(() -> calculateIdpMetaData()`.

Reply:
neat!. missed that one. Updated the PR accordingly